### PR TITLE
botamusique: move to fetchNpmDeps

### PR DESCRIPTION
**Pull request by ajs124**
###### Description of changes
see https://github.com/NixOS/nixpkgs/pull/229910#issuecomment-1535516743

###### Things done
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/audio/botamusique/default.nix [hunk 1]
```diff
@@ -59,17 +53,28 @@ stdenv.mkDerivation rec {
     ./no-runtime-update.patch
   ];
 
+  prePatch = ''
+    postPatchHooks=("''${postPatchHooks[@]/npmConfigHook}")
+  '';
```
Comment (mweinelt) [inline pkgs/tools/audio/botamusique/default.nix:58]: npmSourceRoot when

cc @winterqt @lilyinstarlight

Comment (winterqt) [reply pkgs/tools/audio/botamusique/default.nix:58]: Yeah, that's ugly, yikes.

Give me ~30h.

Comment (winterqt) [reply pkgs/tools/audio/botamusique/default.nix:58]: Hang on, wait, what is this even doing? `npmConfigHook` already runs after the defined `postPatch`...

Comment (mweinelt) [reply pkgs/tools/audio/botamusique/default.nix:58]: Check L67, they are making sure that they can evenlop the hook with pushd/popd to fix the sourceRoot.

Comment (mweinelt) [reply pkgs/tools/audio/botamusique/default.nix:58]: > Hang on, wait, what is this even doing? `npmConfigHook` already runs after the defined `postPatch`...

The pattern removes npmConfigHook from postPatchHooks.

Comment (lilyinstarlight) [reply pkgs/tools/audio/botamusique/default.nix:58]: I think we could probably add an `npmRoot` kinda like `cargoRoot`. Would you like for me to work on that @winterqt?

Comment (winterqt) [reply pkgs/tools/audio/botamusique/default.nix:58]: > I think we could probably add an `npmRoot` kinda like `cargoRoot`. Would you like for me to work on that @winterqt?

That's exactly what @mweinelt is asking for, yeah. I have a branch somewhere that does this, but it's simple enough to implement that if you can do it in the next few hours, go for it. (Else, I will some time tomorrow -- I really don't mind.)

Comment (winterqt) [reply pkgs/tools/audio/botamusique/default.nix:58]: > (alternatively provide pre and post hooks for npmConfigHook)

This is already a thing, FYI -- `prePatch` and `postPatch` both run *before* `npmConfigHook`.

Comment (mweinelt) [reply pkgs/tools/audio/botamusique/default.nix:58]: Heh. The idea was to have something that runs before and after `npmConfigHook`, just like what's currently done in this PR. :laughing:

**Comment by lilyinstarlight** [comment]
Superseded by #230991. Thank you for working on this!

**Pull request closed without merging**